projects/python-client/src/datapane/client/api/report/core.py

```diff
@@ -465,3 +522,107 @@ def preview(
             formatting: Sets the basic report styling
         """
         self._save(self._preview_file.name, open=open, formatting=formatting)
+
+    ############################################################################
+    # Local served reports
+    def build(
+        self,
+        path: str,
+        formatting: t.Optional[ReportFormatting] = None,
+    ) -> None:
+        """Build a report which can be served by a local http server
+
+        Args:
+            path: File path to store the document
+            formatting: Sets the basic report styling
+        """
+
+        pl_path: Path = Path(path)
+        name = pl_path.stem[:127]
```

Comment:
The first 127 characters are taken because the maximum report title length is 127 in the schema rng file.

> Also what happens if the path already exists - do we overwrite, or delete first, or fail?

Right now the behaviour is to overwrite. The `exists_ok=True` inside `bundle_path.mkdir` causes the previous directory to get overwritten if exists